src/ToolBox/SOS/Strike/util.cpp

```diff
@@ -2498,41 +2498,26 @@ BOOL TryGetMethodDescriptorForDelegate(CLRDATA_ADDRESS delegateAddr, CLRDATA_ADD
         return FALSE;
     }
 
-    sos::Object delegateObj = delegateAddr;
+    sos::Object delegateObj = TO_TADDR(delegateAddr);
     int offset;
 
     if ((offset = GetObjFieldOffset(delegateObj.GetAddress(), delegateObj.GetMT(), W("_methodPtrAux"))) != 0)
     {
         CLRDATA_ADDRESS methodPtrAux;
         MOVE(methodPtrAux, delegateObj.GetAddress() + offset);
-        if (methodPtrAux != NULL)
+        if (methodPtrAux != NULL && g_sos->GetMethodDescPtrFromIP(methodPtrAux, pMD) == S_OK)
         {
-            DacpCodeHeaderData codeHeaderData;
-            if (codeHeaderData.Request(g_sos, methodPtrAux) == S_OK)
-            {
-                *pMD = codeHeaderData.MethodDescPtr;
-                return TRUE;
-            }
+            return TRUE;
         }
     }
 
     if ((offset = GetObjFieldOffset(delegateObj.GetAddress(), delegateObj.GetMT(), W("_methodPtr"))) != 0)
     {
         CLRDATA_ADDRESS methodPtr;
         MOVE(methodPtr, delegateObj.GetAddress() + offset);
-        if (methodPtr != NULL)
+        if (methodPtr != NULL && g_sos->GetMethodDescPtrFromIP(methodPtr, pMD) == S_OK)
         {
-            if (g_sos->GetMethodDescPtrFromIP(methodPtr, pMD) == S_OK)
-            {
-                return TRUE;
-            }
-
-            DacpCodeHeaderData codeHeaderData;
```

Comment:
@jkotas, I removed these CodeHeaderData check based on your feedback, but I'm seeing common cases where without it I'm unable to resolve the delegate and with it I am; in those cases ip2md similarly fails.  It sounds like that's not expected?


As an example, I've got a simple ASP.NET Core Web App.  I just created a new one with the template, added this for the index page:
```C#
using System.Threading;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc.RazorPages;

namespace WebApplication1.Pages
{
    public class IndexModel : PageModel
    {
        public async Task OnGetAsync()
        {
            await Task.WhenAll(
                Task.Run(() => SomeMethod()),
                Task.Delay(int.MaxValue));
        }

        public static void SomeMethod() => Thread.Sleep(int.MaxValue);
    }
}
```
ran it, and then attached windbg / sos.  Without this additional code, these checks fail.   With them, they successfully resolve "WebApplication1.Pages.IndexModel+<>c.<OnGetAsync>b__0_0()".

Reply:
Yes, you are right. I see that you have added them in your latest PR already. I got confused by the  DAC API names. The names do not match what one would expect the API to do.